End Grain Options

voltaaage opened the pull request:
WIP: Creates the UI for defining strips to be made with end grains. 
#43
- [x] Create the UI for selecting whether the strip is to be end grain
  <img width="544" alt="screen shot 2016-10-17 at 11 59 36 pm" src="https://cloud.githubusercontent.com/assets/7111237/19467391/cb0c28d6-94c5-11e6-9c5b-33f55b684571.png">
- [x]  Update the strip data when selected
- [x]  Create the ability to update all strips to be End Grain

```
Wizard State Vars
allGrainSelected
allGrainAligned

COMPONENTS
Select all
TRUE
- onClick: call onGrainSelectAll (true)
- isChecked: this.state.allGrainAligned && this.state.allGrainSelected
FALSE
- onClick: call onGrainSelectAll (false)
- isChecked: this.state.allGrainAligned && !this.state.allGrainSelected

Strips
TRUE
- onClick: onEndGrainSelection (true)
    - Update the strip state = true
    - Call checkGrainAlignment
FALSE
- onClick: onEndGrainSelection (false)
    - Update the strip state = false
    - Call checkGrainAlignment

METHODS

Wizard.jsx
onGrainSelectAll (state)
- Update all board strips = true
- Update allGrainSelected = state
- Update allGrainAligned = true

checkGrainAlignment ()
- Check to see that all end grains are the same
- Update allGrainAligned

StripPanel.jsx
onGrainSelection (event)
- Update the strip state = event.currentTarget.value
- Call checkGrainAlignment (May be tricky since this is in the child - maybe call this on wizard's mount)

```
- [ ]  Update textures if the strip is selected to be End Grain

voltaaage committed:
adds end grain section to the strip panel

voltaaage committed:
defaults false for endGrains, updates the strip with attribute with selection

brian-slate commented:
Added the strip end grain props in the API. Let me know when this is ready to review.

voltaaage committed:
WIP - trying to set all end grain

voltaaage committed:
Merge branch 'master' into 43/end-grain-options

voltaaage committed:
adds selected all functionality for end grains

whoa - it works

voltaaage committed:
fixes the issue where you cannot select the end grain for each strip

voltaaage committed:
changes base to new default state for end grains

voltaaage committed:
refactors the alignment check

voltaaage committed:
uses map instead of pop - which broke things

voltaaage committed:
using underscore to determine alignment

voltaaage committed:
going back to the previous approach

the map method in the chain was not extracting the strip properly and I dont see why;

voltaaage commented:
Going to stop wasting time on making a method look prettier b/c the map isn't grabbing the element of the strips collection for some reason. 

Next thing to do is textures - @bmherold, what textures am I trying to replace it with? Are these images available somewhere? I'm looking for the end grain versions of the woods we have

brian-slate commented:
I would say we want on the visual side of this. As long as everything is functional here, I think this is done. Ready to review?

trezy committed:
Remove logging

trezy committed:
Merge branch 'master' into 43/end-grain-options

trezy committed:
Fix broken queryParam splitting
